python/cudf/cudf/_lib/join.pyx

```diff
@@ -222,3 +201,20 @@ cpdef join(Table lhs,
         index = None
     data_ordered_dict = OrderedDict(zip(result_col_names, all_cols_py))
     return Table(data=data_ordered_dict, index=index)
+
+
+def compute_result_col_names(lhs, rhs, how):
+    """
+    Determine the names of the data columns in the result of
+    a libcudf join, based on the original left and right frames
+    as well as the type of join that was performed.
+    """
+    if how in {"left", "inner", "outer", "leftsemi", "leftanti"}:
+        a = lhs._data.keys()
+        if how not in {"leftsemi", "leftanti"}:
+            return [*a, *(k for k in rhs._data.keys() if k not in lhs._data.keys())]
```

Comment:
Would need to do `from itertools import chain` in the file, but I think this reads the intent more clearly.
```suggestion
            return list(chain(a, (k for k in rhs._data.keys() if k not in lhs._data.keys())))
```